planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1112,18 +1129,35 @@ void AvoidanceModule::generateTotalShiftLine(
       sl.shift_line.at(i) = sl.shift_line.at(i - 1);
     }
   }
+
+  if (avoid_lines.empty()) {
+    sl.shift_line_history.push_back(sl.shift_line);
+    return;
+  }
+
+  const auto grad_first_shift_line = (avoid_lines.front().start_shift_length - current_shift) /
+                                     avoid_lines.front().start_longitudinal;
+
+  for (size_t i = avoidance_data_.ego_closest_path_index; i <= avoid_lines.front().start_idx; ++i) {
+    sl.shift_line.at(i) = helper_.getLinearShift(getPoint(path.points.at(i)));
+    sl.shift_line_grad.at(i) = grad_first_shift_line;
+  }
+
   sl.shift_line_history.push_back(sl.shift_line);
 }
 
 AvoidLineArray AvoidanceModule::extractShiftLinesFromLine(ShiftLineData & shift_line_data) const
 {
+  using utils::avoidance::setEndData;
+  using utils::avoidance::setStartData;
+
   const auto & path = avoidance_data_.reference_path;
   const auto & arcs = avoidance_data_.arclength_from_ego;
   const auto N = path.points.size();
 
   auto & sl = shift_line_data;
 
-  const auto getBwdGrad = [&](const size_t i) {
+  const auto backward_grad = [&](const size_t i) {
```

Comment:
Here is also the point you can use lambda function if you think it's good.
```suggestion
  auto calculate_gradient = [&](const size_t i, const bool is_forward) {
    const int direction = is_forward ? 1 : -1;
    if (i == 0 && !is_forward || i == arcs.size() - 1 && is_forward) {
      return sl.shift_line_grad.at(i);
    }

    const double ds = arcs.at(i) - arcs.at(i - direction);
    if (ds < 1.0e-5) {
      return sl.shift_line_grad.at(i);
    }

    return (sl.shift_line.at(i) - sl.shift_line.at(i - direction)) / ds;
  };

```

Reply:
Thanks for your advice!, But I think it is better to separate backward/forward function for readability. I'll fix it in other PR if necessary :+1: